enhancements/etcd/dr-leveraging-ceo-scaling.md

```diff
@@ -0,0 +1,159 @@
+
+## Open Questions [optional]
+
+TODO
+
+## Summary
+
+cluster-etcd-operator (CEO) is an operator that handles the scaling of etcd during cluster
+ bootstrap and regular operation and provisioning etcd dependencies such as TLS certificates.
+
+## Motivation
+
+TODO
+
+
+### Goals
+
+1. Leverage the scale up mechanism of CEO
+
+### Non-Goals
+
+TODO
+
+## Challenges and possible alternatives
+
+A restore operation is a two step process on each node. 
+1. Restoring the data directory with appropriate environment variables with the following operation:
+```
+ETCDCTL_API=3 etcdctl snapshot restore ./path/to/backup.db --name $ETCD_NAME --initial-cluster="$ETCD_NAME=$ETCD_NODE_PEER_URL" --initial-cluster-token openshift-etcd --initial-advertise-peer-urls $ETCD_NODE_PEER_UR  --data-dir="/path/to/new/data-dir"
+```
+2. Starting the etcd pod should be started with the same value of $ETCD_NAME, ETCD_INITIAL_CLUSTER and $ETCD_NODE_PEER_URL as in step 1.
+
+This will allow etcd to populate the data directory with a new cluster ID, and new membership data that includes only 1 member.
+
+Restoring a single member from a backup and scaling up has two distinct problem, they are:
+1. Get the first etcd member running. 
+2. Get the operator to deploy static pods that can decide that they are being restored, so they dont start on the existing data directories.
+
+### Static Pod identifying a restore
+
+Assuming the first member is up, the control plane will be up and operator start running. A user can trigger a new rollout of the etcd pods by running `oc patch forceredpeloymment=reason-time`. New pods will be rolled out on the remaining masters that have to be restored. Before etcd commands on these pods are run, the existing data directory needs to be removed, because they contain the previous cluster state information.
+
+One possible solution is, during the start up pod queries etcd membership, if the pod is an unstarted member, it can be assumed that the data directory present is from previous cluster and moved out.
+
+Other solution is to check the cluster id of the new cluster with the cluster id of the existing data directory, move out the data directory only if they are different. This is a little time consuming to implement and test.
+
+### Getting the first etcd member running
+
+There are two alternative for getting the first etcd member running:
+
+ 1. Admin uses a command to render a bootstrap like pod with recovered data directory. The workflow here would be to:
+    1. move the existing etcd pod yaml out of kubelet manifest directory and move etcd data directory into a different location.
+    2. get the etcd ETCD_NAME and ETCD_PEER_URL values from backed up etcd pod yaml 
+    3. use etcdctl to restore the snapshot 
+    4. start the etcd cert signer
+    5. reuse the CEO binary render command to generate a pod yaml. 
+    6. start the pod on a port different from regular etcd server, peer and metrics ports mount it the new restored data directory
```

Comment:
can we start on the same port?  I wondered about this.  If we used the old name, it would automatically yield.